openeo_processes_dask/process_implementations/comparison.py

```diff
@@ -0,0 +1,140 @@
+from typing import Callable, Optional
+
+import dask.array as da
+import numpy as np
+import xarray as xr
+from numpy.typing import ArrayLike
+from xarray.core.duck_array_ops import notnull
+
+__all__ = [
+    "is_infinite",
+    "is_valid",
+    "is_nodata",
+    "is_nan",
+    "eq",
+    "neq",
+    "gt",
+    "gte",
+    "lt",
+    "lte",
+    "between",
+]
+
+
+def is_infinite(x: ArrayLike):
+    if x is None:
+        return False
+    if (
+        type(x) in [str, list, dict]
+        or type(x) in [np.ndarray, da.core.Array]
+        and x.dtype.kind.lower() in ["u", "s", "o"]
+    ):
+        return False
+    return np.isinf(x)
+
+
+def is_valid(x: ArrayLike):
+    finite = np.logical_not(is_infinite(x))
+    return np.logical_and(notnull(x), finite)
+
+
+def is_nodata(x: ArrayLike):
+    return np.logical_not(is_valid(x))
+
+
+def is_nan(x: ArrayLike):
+    return is_nodata(x)
+
+
+def eq(
+    x: ArrayLike,
+    y: ArrayLike,
+    delta: Optional[float] = None,
+    case_sensitive: Optional[bool] = True,
+):
+    if not type(x) in [np.ndarray, da.core.Array] and not type(y) in [
+        np.ndarray,
+        da.core.Array,
+    ]:
+        if is_nodata(x) or is_nodata(y):
+            return np.nan
+    if x is False or y is False:
+        return False
+    if delta:
+        ar_eq = np.isclose(x, y, atol=delta)
+    elif not case_sensitive:
```

Comment:
why is this an elif? There needs to be a check on whether the array is of dtype string first, only then should case_sensitive be accepted